apps/agentic-chat/src/hooks/useStream.ts

```diff
@@ -96,12 +129,14 @@ export const useStream = (): UseStreamResult => {
           case 'on_chat_model_start': {
             const inputMessages = (data.input?.messages?.[0] ?? []).filter(
               (msg: ChatMessage | null) =>
-                msg?.content?.length && msg._getType() !== 'system'
+                msg?.content?.length &&
+                msg._getType() !== 'system' &&
+                !messages.some((_msg) => _msg.id === msg.id)
```

Comment:
The dedupe guts of rehydration - ensures after rehydration, calling the chain again doesn't include dupe `inputMessages`